packages/block-editor/src/components/use-editor-feature/index.js

```diff
@@ -1,23 +1,51 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
 
+/**
+ * Internal dependencies
+ */
+import { useBlockEditContext } from '../block-edit';
+
 /**
  * Hook that retrieves the setting for the given editor feature.
+ * It works with nested objects using by finding the value at path.
  *
- * @param {string} featureName The name of the feature.
+ * @param {string} featurePath  The path to the feature.
+ * @param {*}      defaultValue Default value to return if not
+ *                              explicitly defined.
  *
  * @return {any} Returns the value defined for the setting.
+ *
+ * @example
+ * ```js
+ * const isEnabled = useEditorFeature( 'typography.dropCap', false );
+ * ```
  */
-export default function useEditorFeature( featureName ) {
+export default function useEditorFeature( featurePath, defaultValue ) {
+	const { name: blockName } = useBlockEditContext();
+	const path = `__experimentalFeatures.${ featurePath }`;
+
 	const setting = useSelect(
 		( select ) => {
+			const { getBlockSupport } = select( 'core/blocks' );
+
+			const blockSupportValue = getBlockSupport( blockName, path );
+			if ( blockSupportValue !== undefined ) {
+				return blockSupportValue;
+			}
+
 			const { getSettings } = select( 'core/block-editor' );
 
-			return getSettings()[ featureName ];
+			return get( getSettings(), path, defaultValue );
```

Comment:
I forgot that we removed the custom ESLint rule we had 😢 But it wouldn't have caught the trickery here anyways.

https://github.com/WordPress/gutenberg/pull/15139#discussion_r278505058

Reply:
I know, but the alternative is to split the string and I'm sure it is comparable to what `get` does internally.

Reply:
> I'm sure it is comparable to what `get` does internally.

As with much of Lodash, you'd be surprised to find out just all it does 😄 

https://github.com/lodash/lodash/blob/master/.internal/stringToPath.js

I'm not actually suggesting any change here, though I do note the inconsistency here if it's something we think should be an ESLint rule in general.

Generally, I think it speaks more to how we've chosen to reference these properties in dot notation.

Reply:
Yes, it was raised by @jorgefilipecosta in https://github.com/WordPress/gutenberg/pull/22291#discussion_r425313407. I feel resistance for using an array as a param in React hooks as it would be used as 2nd param in `useSelect` that is now:
```js
[ blockName, path ]
```

Now I think about it, we could cheat a bit and do:
```js
[ blockName, ...path ]
```

if we opt for a different API for `useEditorFeature`. You both can decide :)

Some alternative options:

1. With a problematic handling for the default value:
```js
const Edit = ( props ) => {
    const isDisabled = ! useEditorFeature( 'typography', 'dropCap', false );
    // ...
};
```

2. With an array as 1st param:
```js
const Edit = ( props ) => {
    const isDisabled = ! useEditorFeature( [ 'typography', 'dropCap' ], false );
    // ...
};
```

Reply:
I mean, there's a certain convenience of the dot notation, I'd grant that. Since it's something of a language hack, I'm not sure just how obvious the syntax is for most people. The array form isn't much better, though 🤷 

It could be just like Lodash where both are technically supported, and the string-form dot notation is casted into the array path. For what it's worth, I implemented a basic [`getPath` implementation](https://github.com/aduth/hpq/blob/master/src/get-path.js) for `hpq` that could be used. _(Though, as I look at it now, it should probably check truthiness of `object` before [operating via `in`](https://github.com/aduth/hpq/blob/c9704368e06be8cfc3d71696dee44d93f7e9dd87/src/get-path.js#L14), to avoid error)_

As you note, `path` could be spread into the dependencies. I have a sense we do this already in some places? I don't really see it as much of a cheat.

Reply:
I want to move on with the proposal as it's blocking other efforts.

The more I think about it, the simplest approach would be to flatten the object structure to avoid the issue in the first place:

```js
const Edit = ( props ) => {
    const isDisabled = ! useEditorFeature( 'dropCap' );
    // ...
};
```
🙂 

Interestingly, it's something that was raised by @youknowriad or @nosolosw in different places but I'm not 100% sure if they were thinking about exactly that.

Reply:
Would it be possible for plugins to add "editor features"? If so, I would think we should probably use namespacing here, e.g. `'wp', 'drop-cap'` or something like that.

Reply:
@ZebulanStanphill since it's using `supports` from the block definition, it's extensible by design. We can discuss it further as the work progresses. It's marked as experimental so it's all open for changes. I didn't think about namespacing so far mostly because there are so many existing features already considered stable so we would need to find migration path.